iceoryx_hoofs/CMakeLists.txt

```diff
@@ -64,86 +64,18 @@ set(${PROJECT_NAME}_DIR ${CMAKE_CURRENT_LIST_DIR}/cmake
     FORCE
 )
 
-#
-########## set util internal target, needed by tests in components ##########
-#
-# TODO: we need something like internal public for internal build dependencies
-# instead of making the whole source folder public for internal dependency resolution
-
 if(BUILD_TEST AND NOT GTest_FOUND)
     find_package(GTest CONFIG REQUIRED)
 endif(BUILD_TEST AND NOT GTest_FOUND)
```

Comment:
Maybe we can find a way to group all the test related CMake Code a bit more together here.
Ideally we have something like:

```cmake
########## hoofs testing ##########
if(BUILD_TEST AND NOT GTest_FOUND)
    find_package(GTest CONFIG REQUIRED)
endif(BUILD_TEST AND NOT GTest_FOUND)

add_subdirectory(testing)
add_subdirectory(test) #taken from the bottom of the file

...
other sections
```

Reply:
The original logic didn't make sense did it?

`iceoryx_hoofs_testing` was only created if gtest was found and a comment mentioned that you could build it without building the unit tests. However, the `find_package Gtest` call was only performed if we did build the unit tests, therefore there wasn't actually a way to build `iceoryx_hoofs_testing` without `BUILD_TESTS=ON`. Thus I merged everything together in the same if clause.

Let me know if that makes sense.